### DISCUSSION TITLE
Can not get past Astroid 3.0.0 with out breaking things

### DISCUSSION THREAD

**Discussion by paurray** [templaza/astroid-framework#469] [Q&A] [answered]
Hello

Maybe this is user error on my part but I can not get past Astroid 3.0.0 with out breaking things.
Maybe some one can give me a tip?

- With my very simple 3 Page Test Site (3 Articles linked to 3 Menus)
- Built in Joomla 3.4.3 locally with MAMP
- PHP version is 8.2.0
- MAMP is Pro and 6.8.1

this is what I do:

- Install Astroid 2.6.6: astroid-package-v2.6.6.Framework-Template.zip

All Good...
...then

- On top of Astroid 2.6.6
- Install astroid-framework-3.0.0.zip
- this works = Framework Version: 3.0.0 installed

All Good...
...but:

- On top of Astroid 3.0.0
- I install astroid-framework-3.0.1.zip
- this does NOT work = Framework Version: 3.0.1 installed and front end gives:

**Oops!! Page not found.**
It seems like the page that you are looking for is no longer here.
ERROR 0 - Call to a member function getRelatedArticles() on null in /Users/name-of-my-computer/Sites/localhost/joomla/J343-A266-304-Frame/libraries/astroid/framework/article.php:266

- On top of Astroid 3.0.1
- I install astroid-framework-3.0.2.zip
- this does NOT work = Framework Version: 3.0.2 installed and front end gives:

**Oops!! Page not found.**
It seems like the page that you are looking for is no longer here.
ERROR 0 - Call to a member function getRelatedArticles() on null in /Users/name-of-my-computer/Sites/localhost/joomla/J343-A266-304-Frame/libraries/astroid/framework/article.php:266

- On top of Astroid 3.0.2
- I install astroid-framework-3.0.3.zip
- this does NOT work = Framework Version: 3.0.3 installed and front end gives:

**Oops!! Page not found.**
It seems like the page that you are looking for is no longer here.
ERROR 0 - Call to a member function getRelatedArticles() on null in /Users/name-of-my-computer/Sites/localhost/joomla/J343-A266-304-Frame/libraries/astroid/framework/library/astroid/Article.php:267

- On top of Astroid 3.0.3
- I install astroid-framework-3.0.4.zip
- this does NOT work = Framework Version: 3.0.4 installed and front end gives:

**Oops!! Page not found.**
It seems like the page that you are looking for is no longer here.
ERROR 0 - Call to a member function getRelatedArticles() on null in /Users/name-of-my-computer/Sites/localhost/joomla/J343-A266-304-Frame/libraries/astroid/framework/library/astroid/Article.php:267

- On top of Astroid 3.0.4
- I install astroid-framework-3.0.5.zip
- this does NOT work = Framework Version: 3.0.5 installed and front end gives:

**Oops!! Page not found.**
ERROR 0 - Call to a member function getRelatedArticles() on null in /Users/name-of-my-computer/Sites/localhost/joomla/J343-A266-304-Frame/libraries/astroid/framework/library/astroid/Article.php:266

Errors are pretty much the same only thing I notice is that sometimes it is Article.php:267 and sometimes Article.php:266

But apart from that no clue! Sorry if this is a stupid question.

thanks for any tips

Paul

**Comment by Chacapamac** [comment]
@paurray 

> ERROR 0 - Call to a member function getRelatedArticles() on null in /Users/name-of-my-computer/Sites/localhost/joomla/J343-A266-304-Frame/libraries/astroid/framework/article.php:266

I guess it is your “Home” page.
Not sure, it seem that something building your front page is not found....

Personally, If you have access to the administration, I will try to rebuild it anew.
Also be sure to assigned the Main menu to your default Home Astroid Layout.

**Comment by sonvnn** [comment, marked as answer]
This issue because this function is not supported in Joomla 4.3. You have to update to Joomla 4.4 to use Astroid 3.

Thanks,
Sonny

**Comment by paurray** [comment]
@sonvnn Perfect thanks. Thanks also @Chacapamac
